### PR TITLE
Add next palindrome problem and take home for FE students

### PR DESCRIPTION
**Pull request by damwhit**

### CODE REVIEW

File: take-homes/fe-take-home.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# NY Times News Reader Take Home Test
+
+You will create a news reader application that pulls articles from the New York Times API.
+
+## Requirements:
+
+At a minimum, you must show:
+
+* A list of articles
```
Comment (dionew1) [inline take-homes/fe-take-home.md:9]: Do you think a minimum/maximum list should be included here or whether or not 'pagination' should be used?

Comment (damwhit) [reply take-homes/fe-take-home.md:9]: For that API I think it's usually less than 100 stories per call. I'm inclined to leave it vague where they could paginate it if they wanted 🤷 . 

One thing I like about this prompt is how it makes them think with a user focus and what is necessary for their less than 6 hour mvp.

@Kalikoze what do you think of the scope of this. If you think it's light for 6 hours we could always add some more requirements too like favoriting or filtering by a tag

File: take-homes/fe-take-home.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+## Requirements:
+
+At a minimum, you must show:
+
+* A list of articles 
+* A "detailed" view for each article (as opposed to summary/list view)
+* Articles in the list must link to the detailed article view you create
+* Some sort of navigation, filter, OR sort
+
+Beyond the base requirements, include any additional elements you'd like, just be mindful of your time. You don't need to go overboard, just consider what might be necessary to achieve MVP so that this news feed is easy-to-use, information is clear, and navigation is painless. Put yourself in a user's shoes and think about what you would need (or want). We aren't looking for UI design, but we would like to see some demonstration of CSS knowledge, and the app should be usable on desktop, tablet, and mobile devices.
```
Comment (Kalikoze) [inline take-homes/fe-take-home.md]: I'm a bit hesitant on telling them "we aren't looking for UI design".  While I think this is a fair point, I do think making a good looking app is important to standing out and taking design ideas from other apps is a key piece to having an effective UX.

File: take-homes/fe-take-home.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+You will create a news reader application that pulls articles from the New York Times API.
+
+## Requirements:
+
+At a minimum, you must show:
+
+* A list of articles 
+* A "detailed" view for each article (as opposed to summary/list view)
+* Articles in the list must link to the detailed article view you create
+* Some sort of navigation, filter, OR sort
```
Comment (Kalikoze) [inline take-homes/fe-take-home.md]: Curious if navigation feels comparable to filter or sort.  If they are implementing navigation, I would imagine it navigates to other built out pages, but that feels like a larger chunk of work.  Maybe navigation could get replaced with search functionality?

File: take-homes/fe-take-home.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+Beyond the base requirements, include any additional elements you'd like, just be mindful of your time. You don't need to go overboard, just consider what might be necessary to achieve MVP so that this news feed is easy-to-use, information is clear, and navigation is painless. Put yourself in a user's shoes and think about what you would need (or want). We aren't looking for UI design, but we would like to see some demonstration of CSS knowledge, and the app should be usable on desktop, tablet, and mobile devices.
+
+## Things we'll be looking for:
+
+* a strong understanding of front-end development fundamentals (primarily JavaScript, but also CSS and HTML)
+* an understanding of React js best practices
+* an understanding of basic usability practices and standards
+* your ability to prioritize for MVP
+* a clear hierarchy of information
+* clean, well thought out code
```
Comment (Kalikoze) [inline take-homes/fe-take-home.md]: Just an idea, similar to what we discussed in our last FE meeting, do some of these "learning goals" feel a bit repetitive?  Maybe we could pick the top 3 here?  Such as:

1.  an understanding of React JS best practices
2. your ability to prioritize for MVP
3. Creating an effective UI and UX?

File: take-homes/fe-take-home.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+Try to limit your total time to around 6 hours on this project. Prioritize what is important to get this feed working and as close to MVP as possible. Please do not attempt to recreate NYTimes.com. We're interested in seeing your ability to write clean code to solve a problem with an understanding of basic UX. We're not looking for fancy UI tricks or hacks.
+
+## NYTimes API Info:
+
+Please use the top stories api located at https://developer.nytimes.com/docs/top-stories-product/1/overview
```
Comment (Kalikoze) [inline take-homes/fe-take-home.md]: Not as important, but maybe this bulletpoint about the API could get moved into the requirements next to `A list of articles` from....the [the top stories api])(link).

**Review by Kalikoze — CHANGES_REQUESTED** [review]
Looks great, thanks for putting this together David!  Palindrome challenge looks great.  I made a few suggestions about the FE take home, but overall I like the openness to it and it gives them an opportunity to be creative here.  (as well as tests them to see if they still do things like a good workflow, testing, READMEs, etc.)

File: take-homes/fe-take-home.md (hunk 1 above)
Comment (Kalikoze) [reply take-homes/fe-take-home.md:9]: I'm fine with the scope of this for now.  This gives them the bare minimum, but I do think it gives them an opportunity to do something to make it stand out whether it's testing or accessibility or adding a new feature.  I'm inclined to try less is more for now.  We can try it out this time and if it feels way too small, we can always add on in our next iteration.

File: take-homes/fe-take-home.md (hunk 3 above)
Comment (damwhit) [reply take-homes/fe-take-home.md]: Fair! And I feel like the "details" view will make some people choose to use a router

File: take-homes/fe-take-home.md (hunk 4 above)
Comment (damwhit) [reply take-homes/fe-take-home.md]: I slimmed these down some. I wonder if having more bullets is more instructive in this case?

File: take-homes/fe-take-home.md (hunk 5 above)
Comment (damwhit) [reply take-homes/fe-take-home.md]: I like it

**Comment by damwhit** [comment]
@dionew1 @Kalikoze thanks for the comments. Should be ready for another looksee!

**Comment by dionew1** [comment]
This looks good to me! @Kalikoze if you are satisfied with the update go ahead and merge the PR. 🙂

**Comment by Kalikoze** [comment]
Looks good to me, let's send it!  🚢

**Pull request merged**